src/compiler/core.ts

```diff
@@ -1489,45 +1562,265 @@ namespace ts {
+            const value = comparer(a, b);
+            return value < 0 ? Comparison.LessThan : value > 0 ? Comparison.GreaterThan : Comparison.EqualTo;
+        }
+
+        function createIntlCollatorStringComparer(locale: string | undefined, caseInsensitive: boolean): Comparer<string> {
+            // Initialize the sort collator on first use
+            let comparer: Comparer<string> = (a, b) => {
+                // Intl.Collator.prototype.compare is bound to the collator. See NOTE in
+                // http://www.ecma-international.org/ecma-402/2.0/#sec-Intl.Collator.prototype.compare
+                comparer = new Intl.Collator(locale, { usage: "sort", sensitivity: caseInsensitive ? "accent" : "variant" }).compare;
+                return comparer(a, b);
+            };
+            return (a, b) => compareWithCallback(a, b, comparer);
+        }
+
+        function createLocaleCompareStringComparer(locale: string | undefined, caseInsensitive: boolean): Comparer<string> {
+            // if the locale is not the default locale (`undefined`), use the fallback comparer.
+            return locale !== undefined ? createFallbackStringComparer(locale, caseInsensitive) :
+                caseInsensitive ? (a, b) => compareWithCallback(a, b, compareCaseInsensitive) :
+                (a, b) => compareWithCallback(a, b, compareCaseSensitive);
+
+            function compareCaseInsensitive(a: string, b: string) {
+                // for case-insensitive comparisons we always map both strings to their
+                // upper-case form as some unicode characters do not properly round-trip to
+                // lowercase (such as `ẞ` (German sharp capital s)).
+                return compareCaseSensitive(a.toLocaleUpperCase(), b.toLocaleUpperCase());
+            }
+
+            function compareCaseSensitive(a: string, b: string) {
+                return a.localeCompare(b);
+            }
+        }
+
+        function createFallbackStringComparer(_locale: string | undefined, caseInsensitive: boolean): Comparer<string> {
+            return caseInsensitive ? (a, b) => compareWithCallback(a, b, compareCaseInsensitive) :
+                (a, b) => compareWithCallback(a, b, compareCaseSensitiveDictionaryOrder);
+
+            function compareCaseInsensitive(a: string, b: string) {
+                // for case-insensitive comparisons we always map both strings to their
+                // upper-case form as some unicode characters do not properly round-trip to
+                // lowercase (such as `ẞ` (German sharp capital s)).
+                return compareCaseSensitive(a.toUpperCase(), b.toUpperCase());
+            }
+
+            function compareCaseSensitive(a: string, b: string) {
+                return a < b ? Comparison.LessThan : a > b ? Comparison.GreaterThan : Comparison.EqualTo;
+            }
+
+            function compareCaseSensitiveDictionaryOrder(a: string, b: string) {

```

Comment:
This is not really case sensitive except in the case of a tie. Is there any case where we *don't* want ties broken? Could we not just always do this?
Actually, this code is probably never hit by tests due to being a fallback -- is this consistent with the behavior of the function it's a fallback for?

Reply:
Yes, this preserves the ordering expectation initially expected by the `compareNavigateToItems` function in _services/navigateTo.ts_.

Reply:
a) Since this is a fallback, shouldn't the others have the same behavior?
b) Maybe `compareNavigateToItems` should just use a case-insensitive comparison?

Reply:
a) Both `Intl.Collator` and correct implementations of `localeCompare` already perform dictionary collation for case-sensitive comparisons.
b) That would result in possible cases of mixed capitalization in results like `Aaa`, `aaa`, `AAa` where we actually want it sorted as `AAa`, `Aaa`, `aaa`.